src/dlgButtonSelect.cpp

```diff
@@ -0,0 +1,233 @@
+
+    buttonSelected = false;
+}
+
+void dlgButtonSelect::slot_pressRadioPressed()
+{
+    pressThresholdSlider->setEnabled(true);
+}
+
+void dlgButtonSelect::slot_releaseRadioPressed()
+{
+    pressThresholdSlider->setDisabled(true);
+}
+
+void dlgButtonSelect::slot_repickButtonClicked()
+{
+    buttonSelected = false;
+    tabWidget->setCurrentIndex(0);
+    deviceEdit->setText("0");
+    buttonEdit->setText("0");
+    pressRadio->click();
+    pressThresholdSlider->setValue(75);
+    repickButton->setDisabled(true);
+}
+
+void dlgButtonSelect::slot_gamepadButtonPress(int deviceId, QGamepadManager::GamepadButton button, double value)
+{
+    lastEventLabel->setText(QString("Device %1: Button %2 was pressed to the %3 value").arg(deviceId).arg(button).arg(value));
+
+    if ( !buttonSelected ){
+        deviceEdit->setText(QString("%1").arg(deviceId));
+        tabWidget->setCurrentIndex(0);
+        buttonEdit->setText(QString("%1").arg(button));
+        buttonSelected = true;
+        repickButton->setEnabled(true);
+        return;
+    }
+
+    if ( button == 12 ){// dpad U
+        pressRadio->click();
+    } else if ( button == 13 ){// dpad D
+        releaseRadio->click();
+    } else if ( button == 15 ){// dpad L
+        if ( pressRadio->isChecked() ){
+            if ( pressThresholdSlider->value() > 4 ){
+                pressThresholdSlider->setValue(pressThresholdSlider->value()-5);
+            } else {
+                pressThresholdSlider->setValue(0);
+            }
+        }
+    } else if ( button == 14 ){// dpad R
+        if ( pressRadio->isChecked() ){
+            if ( pressThresholdSlider->value() < 95)
+            {
+                pressThresholdSlider->setValue(pressThresholdSlider->value()+5);
+            } else {
+                pressThresholdSlider->setValue(99);
+            }
+        }
+    } else if ( button == 0 ){// A
+        accept();
+    } else if ( button == 1 ){// B
+        done(QDialog::Rejected);
+    } else if ( button == 2 ){// X
+        slot_repickButtonClicked();
+        return;
+    }
+
+    buttonSelected = true;
+}
+
+void dlgButtonSelect::slot_gamepadButtonRelease(int deviceId, QGamepadManager::GamepadButton button)
+{
+    //lastEventLabel->setText(QString("Released: dev%1 btn%2").arg(deviceId).arg(button));
+}
+
+void dlgButtonSelect::slot_gamepadAxisEvent(int deviceId, QGamepadManager::GamepadAxis axis, double value)
+{
+    //lastEventLabel->setText(QString("Axis: dev%1 axis%2 val(%3)").arg(deviceId).arg(axis).arg(value));
+    if ( axis == 0 ){
+        axisLeftX = value;
+    } else if ( axis == 1 ){
+        axisLeftY = value;
+    } else if ( axis == 2 ){
+        axisRightX = value;
+    } else if ( axis == 3 ){
+        axisRightY = value;
+    }
+
+    double x, y;
+    QString stick;
+
+    if ( axis == 0 || axis == 1 ){//left stick-- 0:y, 1:x
+        x = axisLeftX;
+        y = axisLeftY;
+        stick = "left";
+    } else {//right stick-- 2:y, 3:x
```

Comment:
The right stick keeps saying button 6 or 7 was pressed when I'm in capture mode.